### DISCUSSION TITLE
Apostrophe 3.21.1: hotfix for backwards compatibility issue with sass

### DISCUSSION THREAD

**Discussion by boutell** [apostrophecms/apostrophe#3786] [Announcements]
Yesterday, `sass` version 1.50.2 was released. This version reports an error when building Apostrophe's admin UI styles, preventing Apostrophe projects from starting up successfully.

Today we've released `apostrophe` version 3.21.1, which temporarily pins our dependency on `sass` to 1.50.1, allowing builds to succeed. In the coming sprint we'll analyze whether the error with sass 1.50.2 is due to an error in our styles, a bug in sass, or a backwards compatibility break in sass, and take appropriate next steps. For now we've done the simplest possible thing to ensure your projects work smoothly.

If you are seeing this error:

```
SassError: RangeError: Invalid value: Not in inclusive range 0..145: -1
```

You can fix it with this command:

```
npm update
```

In a development environment, you might need to also run this command:

```
node app @apostrophecms/asset:clear-cache
```

Thanks for reading.

**Comment by RajanTpss** [comment]
Hello @boutell , I am still facing same as above error even after running the commands.

![image](https://user-images.githubusercontent.com/22762336/172312196-f871e61f-bf57-442b-ad23-d432170b5db1.png)

Please guide me, what else we need to do to remove this error.

Thanks

**Comment by boutell** [comment]
You must "npm update" your project.

If that does not work, check "package.json" and make sure the version number of the dependency on "apostrophe" starts with a "^" and is not locked down (no "^").

**Reply by RajanTpss** [reply]
@boutell , Above is worked for me. Initially this error was comming while we making the build in terminal. Now error is in the browser but build process at console/terminal is successfull. Please have a look at the screenshots : 

![image](https://user-images.githubusercontent.com/22762336/172519072-00f8c0c0-0052-4c54-b964-cc41342a3ea8.png)

Initial Error while making build in console

![image](https://user-images.githubusercontent.com/22762336/172519150-5320b7b2-a940-47d3-8b86-7ed8b0878765.png)

After npm update and remove "^" this from package.json file. Buld in console was working successfully. 

![image](https://user-images.githubusercontent.com/22762336/172519311-adbf1daf-d288-4f6b-82f7-e0b28c0d37d0.png)

But now the error is in browser console as shown in above screenshot.

![image](https://user-images.githubusercontent.com/22762336/172519402-c5d00dfc-e358-488b-8822-0a65c151625d.png)
![image](https://user-images.githubusercontent.com/22762336/172519420-56c82eaf-c026-49c9-a1b2-587c6ac3aa50.png)

Screenshot "package.json" file.

Please guide me to resolve this error. 

Thanks

**Comment by boutell** [comment]
Note that "npm update" and "npm install" are not the same.

**Comment by boutell** [comment]
The "^" is a good thing. You need it.

Without it you are forever locked on apostrophe 3.5.0, which was released several months ago. We have made many fixes since then. Please add the "^" back in front of that and do "npm update".
